### PR TITLE
Add pymethods for better string representation

### PR DESCRIPTION
**Pull request by ketkarameya**
The goal of this PR is to improve the user-experience when developing with piranha's Python API 
* Adds PyO3 getter annotations 
* implements `__str__` and `__repr__` methods (which are called by pyo3 bindings - [ref](https://pyo3.rs/main/class/object#string-representations) )
  * These methods basically call the debug formatter of the object itself. 
      Rust stuff - `format!("{:?}", some_obj)`  triggers `std::fmt::Debug` on `some_obj` - [ref](https://doc.rust-lang.org/std/fmt/index.html)

Added stub file `polyglot_piranha.pyi`. This helps IDEs/editors provide recommendations/completions.

### CODE REVIEW

**Comment by lazaroclapp** [comment]
Not a full review, but, since this adds new functionality, specifically to the python bindings, I would expect to see some related tests 🙂

**Comment by ketkarameya** [comment]
> Not a full review yet, but, since this adds new functionality, specifically to the python bindings, I would expect to see some related tests 🙂

Added tests.

File: src/models/edit.rs [hunk 1]
```diff
@@ -58,3 +62,22 @@ impl Edit {
     }
   }
 }
+
+impl fmt::Display for Edit {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    let replace_range: Range = self.p_match().range();
+    let replacement = self.replacement_string();
+    let replaced_code_snippet = self.p_match().matched_string();
+    let mut edit_kind = "Delete code".red(); //;
```
Comment (lazaroclapp) [inline src/models/edit.rs:71]: What is this `//;` comment for?

File: src/models/piranha_output.rs [hunk 2]
```diff
@@ -16,21 +16,31 @@ use std::path::PathBuf;
...
+use pyo3::{prelude::pyclass, pymethods};
+
+/// A class to represent Piranha's output
 #[derive(Serialize, Debug, Clone, Default)]
 #[pyclass]
 pub struct PiranhaOutputSummary {
+  /// path to the file
   #[pyo3(get)]
   path: String,
+  ///content of the file after all the rewrites
```
Comment (lazaroclapp) [inline src/models/piranha_output.rs:31]:
```suggestion
  /// Content of the file after all the rewrites
```

Presumably, if this is the exported documentation?

File: src/models/piranha_output.rs [hunk 3]
```diff
@@ -16,21 +16,31 @@ use std::path::PathBuf;
...
+
 use super::{edit::Edit, matches::Match, source_code_unit::SourceCodeUnit};
-use pyo3::prelude::pyclass;
+use pyo3::{prelude::pyclass, pymethods};
+
+/// A class to represent Piranha's output
 #[derive(Serialize, Debug, Clone, Default)]
 #[pyclass]
 pub struct PiranhaOutputSummary {
+  /// path to the file
```
Comment (lazaroclapp) [inline src/models/piranha_output.rs:28]:
```suggestion
  /// Path to the file
```

Is there any convention on whether these should be start with lower or uppercase? If not, I'd use "Path" unless you specifically mean the field/variable name.

**Review by lazaroclapp — APPROVED** [review]
LGTM with very minot nits/questions 🙂

**Pull request merged**